src/public/app/widgets/containers/split_note_container.js

```diff
@@ -74,6 +74,47 @@ export default class SplitNoteContainer extends FlexContainer {
         appContext.tabManager.removeNoteContext(ntxId);
     }
 
+    async moveThisNoteSplitCommand({ntxId, isMovingLeft}) {
+        if (!ntxId) {
+            logError("empty ntxId!");
+            return;
+        }
+
+        const contexts = appContext.tabManager.noteContexts;
+
+        const currentIndex = contexts.findIndex(c => c.ntxId === ntxId);
+        const leftIndex = isMovingLeft ? currentIndex - 1 : currentIndex;
+
+        if (currentIndex === -1 || leftIndex < 0 || leftIndex + 1 >= contexts.length) {
+            logError("invalid context!");
+            return;
+        }
+
+        if (contexts[leftIndex].isEmpty() && contexts[leftIndex + 1].isEmpty())
+            // no op
+            return;
+
+        const ntxIds = contexts.map(c => c.ntxId);
+        const mainNtxIds = contexts.map(c => c.mainNtxId);
+
+        this.triggerCommand("noteContextReorder", {
+            ntxIdsInOrder: [
+                ...ntxIds.slice(0, leftIndex),
+                ntxIds[leftIndex + 1],
+                ntxIds[leftIndex],
+                ...ntxIds.slice(leftIndex + 2),
+            ],
+            oldNtxIdsInOrder: ntxIds,
+            mainNtxIdsInOrder: mainNtxIds.map(id => id === ntxIds[leftIndex] ? ntxIds[leftIndex + 1] : id)
```

Comment:
I find these 3 arrays pretty complicated to understand - e.g. the fact that mainNtxIdsInOrder contains duplicated items. I wonder if there's a simpler way to do this.